histogram/src/sparse.rs

```diff
@@ -19,6 +19,114 @@ pub struct SparseHistogram {
     pub count: Vec<u64>,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde-serialize", derive(Serialize, Deserialize))]
+pub struct SparseHistogramRO {
```

Comment:
Let's add a doc comment for the struct. It'd be nice to have something that explains what it is and when you might use it.

Reply:
I'm wondering if calling it an CumulativeSparseHistogram is clearer. I'm not sure I love the short suffix to indicate read-only. ImmutableSparseHistogram might be better if you want to keep the focus on that aspect instead of how it's implemented.

Reply:
I like `CumulativeSparseHistogram`, we can explain why it only supports the read methods in the doc comment.